salt/data-pipeline/bigquery-views.sls

```diff
@@ -0,0 +1,39 @@
+bigquery-view docker image:
+    docker_image.present:
+        # without revision, 'latest' is assumed:
+        # https://docs.saltstack.com/en/latest/ref/states/all/salt.states.dockerng.html#salt.states.dockerng.image_present
+        - load: elifesciences/data-pipeline-bigquery-views
```

Comment:
can this use `project.revision` rather than `latest` like it was proposed for the `git.latest` state `rev` field?
Separately from _can_, _should_ it use that revision? Is this the only versioned artifact (repository, Docker image) that we are deploying to `data-pipeline` servers?

Reply:
There are other projects we are deploying, for example:
- data-pipeline-ejp-to-json-converter
- data-pipeline-ejp-csv-deposit

Currently it's the only Docker image but `data-pipeline-ejp-to-json-converter` should probably follow in the same foot steps.

Reply:
`data-pipeline-ejp-csv-deposit` is a flow support repository. It's NiFi specific stuff and usually wrappers and tiny little scripts and bits of documentation. The name of the repo should probably change to reflect that it's more of a bundle of miscellaneous bits than a single app.

Reply:
or, I could put all these flow support bits into a single repository broken down by flow name (ejp-csv-deposit, ejp-xml-deposit, etc). That might make more sense

Reply:
> probably not. NiFi is general purpose in nature and I suspect the project will pull in new repositories as it needs them.

Yes, I have seen the other repositories and we shouldn't.

> The likelihood of an instance having multiple projects installed on it grows when things get containerised.

I hope to leapfrog this problem by going directly to Kubernetes

> I wonder if we could have a project.revision_map

I too second a different name like `revision_map` to avoid clashes. So far I have used [this trick of storing additional revisions as labels of a main Docker image](https://github.com/elifesciences/builder-base-formula/blob/master/_modules/elife.py#L66-L83) which works when you have one main image and all its dependencies.

Reply:
Btw, from the documentation this shouldn't use `load`:

> Loads a tar archive created with dockerng.load (or the docker load Docker CLI command), and assigns it the specified repo and tag.

Reply:
yes, I think you're right. Another line in the documentation threw me. 

> If neither of the build or load arguments are used, then Salt will pull from the configured registries. 

Reading the docs again (and again) I think what I want is neither load nor build

Reply:
Already updated by @giorgiosironi , cheers.